### PR TITLE
WELD-2224 Cleanup and update of plugins.

### PR DESCRIPTION
**Pull request by manovotn**
Will require a release of our build config project once we merge the PR for checkstyle plugin (https://github.com/weld/build/pull/12).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,20 +22,14 @@
...
       <!-- Force encoding to UTF-8 (bug on OS X) -->
       <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
       <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
       <pdf.name>${project.artifactId}.pdf</pdf.name>
-      <minimum.maven.version>3.0.3</minimum.maven.version>
-      <maven.compiler.source>1.6</maven.compiler.source>
-      <maven.compiler.target>1.6</maven.compiler.target>
+      <minimum.maven.version>3.2.5</minimum.maven.version>
+      <maven.compiler.source>1.7</maven.compiler.source>
+      <maven.compiler.target>1.7</maven.compiler.target>
```
Comment (manovotn) [inline pom.xml]: I am not sure we want/need to increase these versions, but this file should be used in Weld 2.x and 3.x which both require Java 7.
As for Maven, 3.2.5 is the last version of Maven which works with Java 6. I don't think we need to enforce anything more up to date. OTOH I am not sure 3.0.3 is safe for all the plugin updates made here.

File: pom.xml [hunk 2]
```diff
@@ -360,10 +282,6 @@
                               <version>[${minimum.maven.version},)</version>
                            </requireMavenVersion>
                            <requirePluginVersions>
-                              <unCheckedPlugins>
```
Comment (manovotn) [inline pom.xml]: maven-eclipse plugin was removed.
versions-maven plugin is not used here and I haven't found it elsewhere either.

File: pom.xml [hunk 3]
```diff
@@ -166,73 +151,10 @@
...
-               <artifactId>maven-cal10n-plugin</artifactId>
-               <version>${cal10n.plugin.version}</version>
-            </plugin>
-            <plugin>
                <groupId>org.jboss.maven.plugins</groupId>
                <artifactId>maven-jdocbook-plugin</artifactId>
                <version>${jdocbook.plugin.version}</version>
                <extensions>true</extensions>
-               <dependencies>
-                  <dependency>
```
Comment (manovotn) [inline pom.xml]: From my understanding, these docbooks were the previous way of generating documentation. We should now be using asciidoc for this, right?

File: pom.xml [hunk 4]
```diff
@@ -166,73 +151,10 @@
       <pluginManagement>
          <plugins>
             <plugin>
-               <groupId>ch.qos.cal10n.plugins</groupId>
-               <artifactId>maven-cal10n-plugin</artifactId>
```
Comment (manovotn) [inline pom.xml]: This was used for  internationalization in Weld 1.x

File: pom.xml [hunk 5]
```diff
@@ -457,11 +366,6 @@
                <inherited>true</inherited>
             </plugin>
             <plugin>
-               <groupId>org.twdata.maven</groupId>
```
Comment (manovotn) [inline pom.xml]: Didn't find a use of this in our projects.

File: pom.xml [hunk 6]
```diff
@@ -429,15 +347,6 @@
                </executions>
             </plugin>
             <plugin>
-               <groupId>org.apache.maven.plugins</groupId>
-               <artifactId>maven-eclipse-plugin</artifactId>
```
Comment (manovotn) [inline pom.xml]: Retired plugin, not used anymore.

File: pom.xml [hunk 7]
```diff
@@ -575,16 +474,6 @@
             </plugin>
             <plugin>
                <groupId>org.codehaus.mojo</groupId>
-               <artifactId>emma-maven-plugin</artifactId>
-               <version>${emma.plugin.version}</version>
```
Comment (manovotn) [inline pom.xml]: `emma` plugins should be for code coverage but these are pretty outdated and as far as I recall, we are now using `jacoco` for this

File: pom.xml [hunk 8]
```diff
@@ -549,11 +453,6 @@
                <version>${ear.plugin.version}</version>
             </plugin>
             <plugin>
-               <groupId>com.pyx4j</groupId>
-               <artifactId>maven-junction-plugin</artifactId>
```
Comment (manovotn) [inline pom.xml]: Should allow to create symbolic links on Win/*nix systems. Retired, we don't use that anywhere.

File: pom.xml [hunk 9]
```diff
@@ -612,11 +501,6 @@
                <version>${selenium.plugin.version}</version>
             </plugin>
             <plugin>
-               <groupId>org.codehaus.mojo</groupId>
-               <artifactId>failsafe-maven-plugin</artifactId>
```
Comment (manovotn) [inline pom.xml]: `org.codehaus.mojo` is outdated, we use `org.apache.maven.plugins` which is already in our projects.

File: pom.xml [hunk 10]
```diff
@@ -47,57 +41,48 @@
       <manifest.specification.version>1.2</manifest.specification.version>
 
       <!-- dependency versions -->
-      <weld.docbook.version>1.1.1-Beta3</weld.docbook.version>
-      <seam.docbook.version>1.1.0.GA</seam.docbook.version>
-      <jbossorg.docbook.version>1.1.0</jbossorg.docbook.version>
-      <build.config.version>10</build.config.version>
-      <maven.scm.provider.gitexe>1.9.1</maven.scm.provider.gitexe>
+      <build.config.version>12</build.config.version>
```
Comment (manovotn) [inline pom.xml]: NOTE: this is to-be-released version of build-config

**Comment by manovotn** [comment]
Tested by running upgrading `weld-parent` and running  `mvn` `install` and `verify` in:

-  Weld 2.4/3x,
- CDI api 2.x,
- CDI TCK 1.2/2.x

which are all based on this parent. Although many plugins might be triggered only under certain circumstances and profiles so the test was far from exhaustive.

**Comment by mkouba** [comment]
Merged, thanks!

**Pull request closed without merging**